component/src/main/java/org/wso2/extension/siddhi/execution/tensorflow/TensorFlowExtension.java

```diff
@@ -0,0 +1,338 @@
+    private String[] inputVariableNamesArray;
+    private String[] outputVariableNamesArray;
+    private int noOfInputs;
+    private int noOfOutputs;
+    private VariableExpressionExecutor[] inputVariableExpressionExecutors;
+    private Session tensorFlowSession;
+    private SignatureDef signatureDef;
+    private SavedModelBundle tensorFlowSavedModel;
+
+    @Override
+    protected List<Attribute> init(AbstractDefinition abstractDefinition, ExpressionExecutor[] expressionExecutors,
+                                   ConfigReader configReader, SiddhiAppContext siddhiAppContext) {
+
+        final int minConstantParams = 3;
+
+        //Checking if at least minimum number of constant params are present in the query
+        if (attributeExpressionLength < minConstantParams) {
+            String message = "Insufficient number of parameters. Query should have at least 5 constant parameters " +
+                    "and appropriate number of variable parameters but given " + attributeExpressionLength;
+            logger.error(siddhiAppContext.getName() + message);
+            throw new SiddhiAppCreationException(message);
+        }
+
+        //expressionExecutors[0] --> absolute path to model
+        String modelPath;
+        if (!(attributeExpressionExecutors[0] instanceof ConstantExpressionExecutor)) {
+            throw new SiddhiAppCreationException("1st query parameter is the absolute path to model which has to be " +
+                    "constant but found " + this.attributeExpressionExecutors[0].getClass().getCanonicalName());
+        }
+        if (attributeExpressionExecutors[0].getReturnType() == Attribute.Type.STRING) {
+            modelPath = (String) ((ConstantExpressionExecutor) attributeExpressionExecutors[0]).getValue();
+        } else {
+            throw new SiddhiAppCreationException("1st query parameter is the absolute path to model which has to " +
+                    "be of type String but found " + attributeExpressionExecutors[0].getReturnType());
+        }
+
+        //loading the saved model
+        final String servingTag = "serve";
+        tensorFlowSavedModel = SavedModelBundle.load(modelPath, servingTag);
+        tensorFlowSession = tensorFlowSavedModel.session();
+
+        //Loading signatureDef
+        final String defaultServingSignatureDefKey = "serving_default";
+        try {
+            signatureDef =
+                    MetaGraphDef.parseFrom(tensorFlowSavedModel.metaGraphDef())
+                            .getSignatureDefOrThrow(defaultServingSignatureDefKey);
+        } catch (InvalidProtocolBufferException e) {
+            throw new SiddhiAppCreationException("Error while reading signature def." + e.getMessage(), e);
+        }
+
+        noOfInputs = signatureDef.getInputsCount();
+        noOfOutputs = signatureDef.getOutputsCount();
+
+        //Checking if the specified number of inputs are given
+        final int noOfQueryParams = 1 + 2 * noOfInputs + noOfOutputs;
```

Comment:
Shall we replace (1+2) this with minConstants

Reply:
that is 1 + (2 x noOfInputs)

Reply:
Shall we add a comment describing what 1 and 2 are

Reply:
added